src/components/repo-list-item.component.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import { StyleSheet, Text, View } from 'react-native';
+import { ListItem, Icon } from 'react-native-elements';
+
+import { emojifyText, abbreviateNumber } from 'utils';
+import { colors, languageColors, fonts, normalize } from 'config';
+
+type Props = {
+  repository: Object,
+  showFullName: boolean,
+  navigation: Object,
+};
+
+const styles = StyleSheet.create({
+  wrapper: {
+    marginTop: 5,
+    marginBottom: 5,
+    marginLeft: 5,
+  },
+  titleWrapper: {
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+  title: {
+    color: colors.primaryDark,
+    ...fonts.fontPrimarySemiBold,
+  },
+  privateIconContainer: {
+    marginLeft: 6,
+  },
+  description: {
+    color: colors.primaryDark,
+    ...fonts.fontPrimaryLight,
+  },
+  extraInfo: {
+    flexDirection: 'row',
+    flex: 1,
+    paddingTop: 5,
+  },
+  extraInfoSubject: {
+    color: colors.greyDark,
+    paddingLeft: 3,
+    paddingTop: 2,
+    marginRight: 15,
+    fontSize: normalize(10),
+    ...fonts.fontPrimary,
+  },
+  repositoryContainer: {
+    justifyContent: 'center',
+    flex: 1,
+  },
+});
+
+const renderTitle = (repository, showFullName) =>
+  <View style={styles.wrapper}>
+    <View style={styles.repositoryContainer}>
+      <View style={styles.titleWrapper}>
+        <Text style={styles.title}>
+          {showFullName ? repository.fullName : repository.shortName}
+        </Text>
+        {repository.private &&
+          <View style={styles.privateIconContainer}>
+            <Icon
+              size={16}
+              name="lock"
+              type="octicon"
+              color={colors.greyDarkest}
+            />
+          </View>}
+      </View>
+      <Text style={styles.description}>
+        {emojifyText(repository.description)}
+      </Text>
+    </View>
+    <View style={styles.extraInfo}>
+      <Icon
+        containerStyle={styles.extraInfoIcon}
+        name="star"
+        type="octicon"
+        size={15}
+        color={colors.greyDark}
+      />
+
+      <Text style={styles.extraInfoSubject}>
+        {abbreviateNumber(repository.countStargazzers)}
+      </Text>
+
+      <Icon
+        containerStyle={styles.extraInfoIcon}
+        name="repo-forked"
+        type="octicon"
+        size={15}
+        color={colors.greyDark}
+      />
+
+      <Text
+        style={[styles.extraInfoSubject, { paddingLeft: 0, marginRight: 13 }]}
+      >
+        {abbreviateNumber(repository.countForks)}
+      </Text>
+
+      {repository.language !== null &&
+        <Icon
+          containerStyle={styles.extraInfoIcon}
+          name="fiber-manual-record"
+          size={15}
+          color={languageColors[repository.language]}
+        />}
+
+      <Text style={styles.extraInfoSubject}>
+        {repository.language}
+      </Text>
+    </View>
+  </View>;
+
+export const RepoListItem = ({ repository, showFullName, navigation }: Props) =>
```

Comment:
Maybe use full name `RepositoryListItem`, because there is already a `RepositoryList`?

Reply:
Yup yup.

Again, naming/imports paths were not really polished, and we need to focus on more critical architectural stuff for the store ;)

Did you get a chance to watch the Redux store being populated? What do you think about the schemas thing?

Reply:
Yes, I looked through the debugger, it's simple and understandable. Results are saved (cached), very good with our navigation. And most importantly, you can refuse now from `reducers` and `actions`, delete them! I could not imagine, but it turned out to be not so difficult, all ingenious is simple :smile: 

Of course, we still need to test, but in general it's very good.

Regarding the scheme, I thought it would be a general (basic) scheme for each entity, because we will have a own scheme for REST (v3) and GraphQL (v4)? I wanted them to follow the basic scheme, where by the way the types of fields would be spelled out eg.

Reply:
Completely. The basic scheme would also apply to future Bitbucket & Gitlab schema.
I wanted to use flow to add type checking, but I'm unable to get it to work correctly

Reply:
To implement Flow I think it may be already in a separate PR?

But as I understand, in `normalizr` there is no way to do the basic schema?

Reply:
We normally already have flow configured in the repo (.flowconfig).

I'm not sure I understand your question about the basic schema for normalizr.

Currently, I implement a Schema by entity. In that schema, I handle  the JSON => entity attributes mapping. 
As the JSON structure from GraphQL will probably be different, we will have to implement new Schema, that will transform GraphQL JSON to the same entity structure we have for REST.

Reply:
> As the JSON structure from GraphQL will probably be different, we will have to implement new Schema, that will transform GraphQL JSON to the same entity structure we have for REST.

I meant it, the schema fields will be the same, but because REST and GraphQL return them in different ways (`avatar_url` and `avatarUrl`), I wanted to have a basic schema where all these fields were specified, and they were all in these two schemes (REST and GraphQL).

Reply:
I don't think it's possible with Normalizr, that's why I was counting on some sort of type checker instead